backend/users/signals.py

```diff
@@ -28,3 +28,10 @@ def ensure_admin_email(sender, instance, created, **kwargs):
             print(f'Automatically verified email {instance.email} for {instance}')
         except Exception as e:
             print('Failed to automatically verify admin email', e, sep='\n')
+
+@receiver(post_save, sender=CustomUser)
+def create_user_profile(sender, instance, created, **kwargs):
+    if created:
+        UserProfile.objects.get_or_create(
+            user=instance
+        )
```

Comment:
This combination makes no sense. If a `User` is created, there should not be a `UserProfile` because it's a 1-to-1 relation. Cases where the `get` part is executed instead are clear bugs and should not be passed silently.
```suggestion
    if created:
        UserProfile.objects.create(user=instance)
```